### PR TITLE
Destination Redshift: Fixed maximum record size for SUPER type

### LINKED ISSUE #12680: Redshift Destination SUPER type max size limit too low
## Environment
- **Airbyte version**:  0.36.6-alpha
- **OS Version / Instance**: AWS Linux 2
- **Deployment**: Kubernetes
- **Source Connector and version**: N/A
- **Destination Connector and version**: Redshift 0.3.32
- **Severity**: Medium
- **Step where error happened**: Sync Job

## Current Behavior
Data larger than 65535 is dropped. [Docs](https://docs.airbyte.com/integrations/destinations/redshift/#data-size-limitations), [source](https://github.com/airbytehq/airbyte/blob/60cbf0d0d943de344c9c413fe9f47fb84674fbf0/airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftSqlOperations.java#L97)

## Expected Behavior
AWS [docs](https://docs.aws.amazon.com/redshift/latest/dg/r_SUPER_type.html) for SUPER state a 1MB max size. `isValidData` needs to handle varchar vs SUPER data type differently.

## Logs

<details>
<summary>LOG</summary> 

```
2022-05-06 02:25:16 [43mdestination[0m > 2022-05-06 02:25:16 [32mINFO[m i.a.i.b.FailureTrackingAirbyteMessageConsumer(close):65 - Airbyte message consumer: succeeded.
2022-05-06 02:25:16 [43mdestination[0m > 2022-05-06 02:25:16 [33mWARN[m i.a.i.d.b.BufferedStreamConsumer(lambda$close$0):166 - A total of 7750 record(s) of data from stream AirbyteStreamNameNamespacePair{name='<name>', namespace='<db>'} were invalid and were ignored.
2022-05-06 02:25:16 [43mdestination[0m > 2022-05-06 02:25:16 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(close):170 - executing on success close procedure.
2022-05-06 02:25:16 [43mdestination[0m > 2022-05-06 02:25:16 [32mINFO[m i.a.i.d.r.InMemoryRecordBufferingStrategy(lambda$flushAll$1):84 - Flushing 

```

</details>

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?

Remove this with your answer.

### PR DESCRIPTION
**Pull request by adam-bloom**
## What
See #12680: now that airbyte uses SUPER instead of VARCHAR, the max data size needs to be increased.

## How
Changes the constant. #12064 changes the type for all operations (not configurable), so we don't need to be clever.

https://docs.aws.amazon.com/redshift/latest/dg/r_SUPER_type.html

Also see https://docs.aws.amazon.com/redshift/latest/dg/limitations-super.html - the VARCHAR check is retained, but only used for fields within the SUPER object instead of the object overall.

## Recommended reading order
N/A

## 🚨 User Impact 🚨
N/A

## Pre-merge Checklist

<summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@adam-bloom can you sign the CLA? 
@grubberr can you review this contribution?

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java [hunk 1]
```diff
@@ -92,9 +94,26 @@ public void insertRecordsInternal(final JdbcDatabase database,
...
+    // check VARCHAR limits for VARCHAR fields within the SUPER object
+    Map<String, Object> dataMap = Jsons.flatten(data);
+    for (Object value : dataMap.values()) {
+      if (value instanceof String) {
+        String stringValue = (String) value;
+        final int stringDataSize = stringValue.getBytes(StandardCharsets.UTF_8).length;
+        isValid = stringDataSize <= REDSHIFT_VARCHAR_MAX_BYTE_SIZE;
+        if (!isValid) {
+          break;
+        }
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java:113]:
```suggestion
     if (value instanceof String stringValue) {
        final int stringDataSize = stringValue.getBytes(StandardCharsets.UTF_8).length;
        isValid = stringDataSize <= REDSHIFT_VARCHAR_MAX_BYTE_SIZE;
        if (!isValid) {
          break;
        }
```

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java [hunk 2]
```diff
@@ -92,9 +94,26 @@ public void insertRecordsInternal(final JdbcDatabase database,
 
   @Override
   public boolean isValidData(final JsonNode data) {
+    boolean isValid = true;
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java:97]: I believe it is not necessary to init the isValid variable.

**Comment by adam-bloom** [comment]
> @adam-bloom Thanks for the contribution, good catch. Please, follow the Pre-merge checklist. Nothing critical from my side.

@alexandertsukanov what parts of the checklist specifically? I attempted to run the integration tests, but they require AWS credentials for the redshift destination tests. Are these expected to be runnable in any AWS account (and I should just use my own creds) or do these need to be run by someone on the airbyte team? Otherwise, I think it's just adding a changelog entry, correct?

Thanks for the few syntax suggestions - I'll clean those up along with adding the changelog entry.

**Comment by marcosmarxm** [comment]
@adam-bloom please update the connector version in the Dockerfile please

**Comment by adam-bloom** [comment]
> @adam-bloom please update the connector version in the Dockerfile please

@marcosmarxm done

**Comment by alexandertsukanov** [comment]
Looks like a lot of tests fails with https://github.com/airbytehq/airbyte/runs/6497341731?check_suite_focus=true
`  2022-05-18 21:34:36 INFO i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$0):61 - Caused by: com.amazon.support.exceptions.ErrorException: [Amazon](500310) Invalid operation: password authentication failed for user "awsuser";`
Could be forked repository the root cause of that?

Also, I see failed tests here 
`RedshiftInsertDestinationAcceptanceTest > testSyncVeryBigRecords() FAILED
    org.opentest4j.AssertionFailedError: expected: <6> but was: <7>
        at app//org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at app//org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:527)
        at `

**Comment by adam-bloom** [comment]
@alexandertsukanov @marcosmarxm would one of you be able to trigger a re-test? I think I was able to figure out the issue (and find another unrelated one while I was at it!)

**Comment by alexandertsukanov** [comment]
> @alexandertsukanov @marcosmarxm would one of you be able to trigger a re-test? I think I was able to figure out the issue (and find another unrelated one while I was at it!)

I have run again

**Comment by alexandertsukanov** [comment]
Looks good now. 
CC: @marcosmarxm, @adam-bloom

**Comment by alexandertsukanov** [comment]
@adam-bloom could you bump the version of the connector after I will approve. Thanks.

**Comment by adam-bloom** [comment]
> @adam-bloom could you bump the version of the connector after I will approve. Thanks.

@alexandertsukanov where does the version still need to be bumped? The changelog and dockerfile are already updated.

**Comment by alexandertsukanov** [comment]
> > @adam-bloom could you bump the version of the connector after I will approve. Thanks.
> 
> @alexandertsukanov where does the version still need to be bumped? The changelog and dockerfile are already updated.

sorry, my bad, looks good so far

**Review by alexandertsukanov — APPROVED** [review]

**Comment by marcosmarxm** [comment]
> @adam-bloom could you bump the version of the connector after I will approve. Thanks.

I'll test the auto-bumper if it's working

**Comment by adam-bloom** [comment]
Looks like something failed - it thinks 0.3.35 is already used? Is that also on another branch somewhere and this needs to go up one more?

**Comment by alexandertsukanov** [comment]
I see next
image
Looks like version was published by some PR, but I don't understand why Dockerfile has another

**Comment by alexandertsukanov** [comment]
here is the PR where the connector was published https://github.com/airbytehq/airbyte/pull/12820 with 35 version
So, I think I can publish this one after mentioned will be merged.

**Review by marcosmarxm — APPROVED** [review]
thanks @adam-bloom

**Pull request merged**